### PR TITLE
WFNEWS-2411 Use CapacitorHttp on mobile to avoid CORS issues in highlights widget

### PR DESCRIPTION
**Pull request by ssylver93**
The standalone iOS app is not returning any results from the WordPress API due to CORS errors when using Angular's HttpClient. This change will detect if the Capacitor native platform is being used and fetch the blog posts for the highlights widget using CapacitorHttp if so.

### CODE REVIEW

**Review by yzlucas — APPROVED** [review]

File: client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts [hunk 1]
```diff
@@ -139,4 +141,22 @@ export class HighlightsWidgetComponent implements OnInit {
       fireCentres: fireCentreTags.map(tag => tag.name)  // Array of fire centre names
     };
   }
+
+  // Use CapacitorHttp if on mobile to avoid CORS issues
+  private processRequest<T>(url: string): Observable<T> {   
+    if (Capacitor.isNativePlatform()) {
```
Comment (sukhpalp) [inline client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts]: ~Is there a reason to not use CapacitorHttp? Could we not use it on non native as well?~

I misunderstood how this library worked. Please disregard this comment. The one below still applies though, and I think we should keep this logic in some sort of HttpService.

File: client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts [hunk 2]
```diff
@@ -139,4 +141,22 @@ export class HighlightsWidgetComponent implements OnInit {
       fireCentres: fireCentreTags.map(tag => tag.name)  // Array of fire centre names
     };
   }
+
+  // Use CapacitorHttp if on mobile to avoid CORS issues
+  private processRequest<T>(url: string): Observable<T> {
```
Comment (agordon-vivid) [inline client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts]: I assume we're using capacitorhttp elsewhere already?  

Is this function something that should live in a more general context?  Will other components need to use this same technique?

File: client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts (hunk 1 above)
Comment (sukhpalp) [reply client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts]: If we cannot use the CapacitorHttp for non native HTTP call, I would suggest that we write a wrapping service that switches between the two HTTP providers in a central service, as opposed to spreading that logic throughout the code.

File: client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts (hunk 2 above)
Comment (agordon-vivid) [reply client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts]: The more I think about this, the more I'd like to see logic like this in a service class.

**Review by agordon-vivid — CHANGES_REQUESTED** [review]
Plese move request logic to a service class.

File: client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts (hunk 1 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts]: Thanks, I moved the getRequest function to the common utility service now, and added some formatting/linting changes in the files I was working on.

File: client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts (hunk 2 above)
Comment (ssylver93) [reply client/wfnews-war/src/main/angular/src/app/components/dashboard-component/widgets/highlights-widget/highlights-widget.component.ts]: Thanks, I moved the getRequest function to the common utility service now, and added some formatting/linting changes in the files I was working on.

**Review by agordon-vivid — APPROVED** [review]

**Review by sukhpalp — APPROVED** [review]

**Pull request merged**